<launch>
    <arg name="environment" default="house" />
    <arg name="metric_map" default="$(find learning_toponav)/maps/$(arg environment)/metric.yaml" />
    <arg name="topomap" default="$(find learning_toponav)/maps/$(arg environment)/topomap.tpg" />
    <arg name="config_yaml" default="$(find learning_toponav)/param/config.yaml" />
    <arg name="rviz_config" default="-d $(find learning_toponav)/rviz/testing.rviz" />
    <arg name="gazebo_gui" default="false" /> <!-- ACTIVATE GAZEBO GUI -->
    <arg name="rqt_console" default="false" /> <!-- LAUNCH RQT CONSOLE -->

    <!-- GAZEBO -->
    <include file="$(find learning_multirobot)/launch/includes/main_gazebo.launch">
        <arg name="world_name" value="$(find learning_toponav)/worlds/$(arg environment).world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="$(arg gazebo_gui)"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!--gzserver killer-->
	<node pkg="my_fun" type="gz_killer.py" name="gz_killer" output="screen" />

    <!-- map server -->
    <node pkg="map_server" type="map_server" name="map_loader" args="$(arg metric_map)" />

    <!--toponodes publisher-->
    <node pkg="learning_toponav" type="toponodes_publisher.py" name="toponodes_publisher"
          args="--topomap $(arg topomap) --yaml $(arg config_yaml)"
    />

    <!--robot spawner-->
    <node pkg="learning_toponav" type="robot_spawner.py" name="robot_spawner"
          args="--yaml $(arg config_yaml)" output="screen"
    />

    <!--topoplanner-->
    <arg name="adjlist" value="$(find learning_toponav)/maps/$(arg environment)/adjlist.txt" />
    <arg name="planner_logs" default="true" />
    <node pkg="learning_toponav" type="topoplanner.py" name="topoplanner"
          args="--adjlist $(arg adjlist) --environment $(arg environment)
                --yaml $(arg config_yaml) --logging $(arg planner_logs)"
          output="screen" required="true"
    />

    <!--destinations debugger-->
    <arg name="dest_debug" default="false" />
    <node pkg="learning_toponav" type="destinations_debugger.py" name="destinations_debugger"
          args="--yaml $(arg config_yaml)" if="$(arg dest_debug)"
    />

    <!-- RVIZ -->
    <node name="rviz" pkg="rviz" type="rviz" args="$(arg rviz_config)" />

    <!--rqt_console-->
    <node pkg="rqt_console" type="rqt_console" name="rqt_console" if="$(arg rqt_console)" />
</launch>